src/test/java/com/eighthlight/fabrial/test/server/HttpHEADAcceptanceTest.java

```diff
@@ -0,0 +1,32 @@
+package com.eighthlight.fabrial.test.server;
+
+import com.eighthlight.fabrial.http.Method;
+import com.eighthlight.fabrial.http.Request;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URI;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class HttpHEADAcceptanceTest extends TcpServerIntegrationTest {
+  @BeforeEach
+  void startAndConnect() throws IOException {
+    server.start();
+    client.connect();
+  }
+
+  @Test
+  void simpleHEADRequest() throws Throwable {
+    Request req = new Request("1.1", Method.HEAD, new URI("/test")) ;
```

Comment:
Interesting, I'll try to repro. Otherwise, I still stand by my use of acceptance tests here, as it's way faster than me running `curl` on the command line, modifying the code, re-running the app, etc. Of course, I'm still in the process of adding more unit tests 😉

Reply:
Well sure, manual tests << integration/acceptance/e2e tests << unit tests.

Reply:
Well sure, manual tests << integration/acceptance/e2e tests << unit tests.

Reply:
Acceptance tests passing locally for me: `./gradlew testAcceptance -i`. I'm guessing that it's either an issue with the port being used on your machine, or (IME, infrequent) race conditions between the server starting and the client's connection attempt. If it proves to be flakey in CI as well, I'll DRY up the setup logic so it retries the client connection before running tests.

Reply:
Sorry, this is actually an integration test (rename commit incoming). The whole suite is passing locally for me. There are some error logs written during the test, but as long as it shows "passing" it should be fine.

Reply:
I'm seeing the issue when running it via IntelliJ "all tests", depending on the ordering, and I am now seeing that it's intermittent. 

As a smaller, more focused case, if I highlight those two test classes together (`AppAcceptanceTest` and `HttpHeadIntegrationTest`), and run those (in that order), the second one sometimes fails with the error I mentioned. Perhaps that will allow you to reproduce?

I do see the suite passing when I run the separate suites that CI runs, for what it's worth - in this case those tests never run quickly enough back-to-back to uncover the shared resource conflict.

When running tests in the IDE, running "all tests" tends to be a default for a lot of folks, so even if it passes in CI, I think this is worth addressing. One practice I've seen that would have solved the *direct* bug for me would be to use this separate `test-integration` source directory that you have in your `build.gradle` (but I think aren't using?) for slow tests. This wouldn't solve the underlying root cause (competition for a single shared resource, the port), but would make it less likely to encounter.